src/main/java/emissary/core/IBaseDataObjectXmlCodecs.java

```diff
@@ -620,6 +620,36 @@ public void encode(final List<Map<String, byte[]>> values, final Element parentE
         }
     }
 
+    public static final ElementEncoder<Map<String, byte[]>> SHA256_STRING_BYTE_ARRAY_ENCODER = new Sha256StringByteArrayEncoder();
+
+    private static class Sha256StringByteArrayEncoder implements ElementEncoder<Map<String, byte[]>> {
+        @Override
+        public void encode(final List<Map<String, byte[]>> values, final Element parentElement, final String childElementName) {
+            for (final Map<String, byte[]> value : values) {
+                for (final Entry<String, byte[]> view : value.entrySet()) {
+                    final Element metaElement = new Element(IbdoXmlElementNames.VIEW);
+
+                    parentElement.addContent(metaElement);
+                    metaElement.addContent(preserve(protectedElement(IbdoXmlElementNames.NAME, view.getKey())));
+                    metaElement.addContent(preserve(protectedElementHash(IbdoXmlElementNames.VALUE, view.getValue())));
+                }
+            }
+        }
+
+        private static Element protectedElementHash(final String name, final byte[] bytes) {
```

Comment:
Should this method be moved out of this nested type and mirror `protectedElementBase64(..)`?